pkg/storage/storage.go

```diff
@@ -264,3 +264,37 @@ func GetMountPath(client *occlient.Client, applicationName string, componentName
 	}
 	return mPath
 }
+
+// Mount mounts the given storage to the given component
+func Mount(client *occlient.Client, path string, storageName string, componentName string, applicationName string) error {
+	storageComponent, err := getComponentNameFromStorageName(client, storageName)
+	if err != nil {
+		return errors.Wrap(err, "unable to get the component name associated with the storage")
+	}
+	if storageComponent != "" {
+		return fmt.Errorf("the given storage is already mounted to the component %v", storageComponent)
+	}
+	pvc, err := client.GetPVCFromName(generatePVCNameFromStorageName(storageName))
+	if err != nil {
+		return errors.Wrap(err, "unable to get the pvc from the storage name")
+	}
+	err = client.UpdatePVCLabels(pvc, storagelabels.GetLabels(storageName, componentName, applicationName, true))
```

Comment:
PVC labels should be updated only after the volume was successfully added to the DC.
This should be the last thing that is done in this function.